### PR TITLE
[rc] Adjustments to `skyux upgrade`

### PR DESCRIPTION
**Pull request by Blackbaud-SteveBrush**
- Add support for installing the preferred versions of `typescript`, `ts-node`, and `zone.js` (this will allow us to enforce specific versions of those packages, instead of relying on the user to find the "best" match).
- Bugfix: allow installing pre-releases versions of peer dependencies (currently only the `latest` versions of the peers are installed).
- Updated/cleaned-up console logs throughout.

### CODE REVIEW

File: lib/app-dependencies.js [hunk 1]
```diff
@@ -94,62 +121,82 @@ async function addSkyPeerDependencies(dependencies) {
...
+          );
           break;
         default:
-          if (dependencies[packageName] === versionNumber) {
-            logger.info(`Package ${packageName} already on latest version (${versionNumber})`);
+          if (
+            /^[0-9]+\./.test(currentVersion) ||
+            /-(rc|alpha|beta)\./.test(currentVersion)
+          ) {
+            versionRange = `^${currentVersion.replace('^', '').replace('~', '')}`;
```
Comment (Blackbaud-BobbyEarl) [inline lib/app-dependencies.js]: Why is this necessary?

Comment (Blackbaud-SteveBrush) [reply lib/app-dependencies.js]: Great catch; I wasn't taking into consideration how many formats a version range can take. Per your suggestion, I'm now using the `semver` package.
